packages/controller/src/tasks/syncTenants.ts

```diff
@@ -21,7 +21,58 @@ import { KubeConfig } from "@kubernetes/client-node";
 import { set as updateTenants, Tenants } from "@opstrace/tenants";
 import dbClient, { ClientResponse } from "../dbClient";
 import { State } from "../reducer";
-import { log, SECOND } from "@opstrace/utils";
+import { log, SECOND, httpcl } from "@opstrace/utils";
+import { Response as GotResponse } from "got";
+
+function* setDefaultAlertmanagerConfigIfEmpty(tenant: string) {
+  const res: GotResponse<string> = yield httpcl(
+    `http://alertmanager.cortex.svc.cluster.local/api/v1/alerts`,
+    {
+      method: "GET",
+      headers: {
+        "X-Scope-OrgID": tenant
+      },
+      throwHttpErrors: false
+    }
+  );
+
+  if (res.statusCode > 499) {
+    // Network error, don't potentially overwrite existing config
+    log.error(
+      `could not read alertmanager config for tenant: ${tenant}, got resp: ${res.statusCode}, ${res.statusMessage}`
+    );
```

Comment:
The condition `if (res.statusCode > 499)` means "handle 5xx responses". What we typically want to do in those cases is to log the response body in a responsible fashion; to see the error detail emitted by the remote end.

I think `${res.statusMessage}` is just a hard-coded translation of the status code, and not a peek into the response body. Maybe log `res.body.slice(0, 500)`.

Reply:
Yes, good point and great suggestion!